uaclient/security_status.py

```diff
@@ -31,18 +28,23 @@ class UpdateStatus(Enum):
     UNAVAILABLE = "upgrade_unavailable"
 
 
-def list_esm_for_package(package: apt_package.Package) -> List[str]:
-    esm_services = []
-    for origin in package.installed.origins:
-        if (origin.origin, origin.archive) == SERVICE_TO_ORIGIN_INFORMATION[
-            "esm-infra"
-        ]:
-            esm_services.append("esm-infra")
-        if (origin.origin, origin.archive) == SERVICE_TO_ORIGIN_INFORMATION[
-            "esm-apps"
-        ]:
-            esm_services.append("esm-apps")
-    return esm_services
+def get_origin_for_package(package: apt_package.Package) -> str:
+    available_origins = package.installed.origins
+    if len(package.installed.origins) == 1:
+        if package.installed == package.candidate:
+            return "unknown"
+        available_origins = package.candidate.origins
```

Comment:
These special cases for when the installed origins == 1 aren't obvious to someone unfamiliar with the apt internals. Can you leave a comment explaining the "unknown" case and the "available_origins" special case?

Reply:
Sure